app/registration/api.py

```diff
@@ -44,7 +44,44 @@ def api_register():
         data = registration.blob,
     )
 
-@registration_blueprint.route('/api/order/token')
+@registration_blueprint.route('/api/priorities', methods=['GET', 'POST'])
+@oauth.require_oauth('registration_priorities')
 @token_auth.login_required
-def validate_token():
-    return jsonify(g.uni.name)
+def api_registration_priorities():
+    if request.method == 'POST' \
+        and request.headers.get('Content-Type') == 'application/json':
+        req = request.get_json()
+        if not req:
+            abort(403)
+            return ''
+
+        new_priority = len(req['priority'])
+
+        registrations = Registration.query.filter_by(uni_id = g.uni.id)
+        for registration in registrations:
+            if registration.id in req['confirmed']:
+                registration.confirmed = True
+                try:
+                    registration.priority = req['order'].index(reg_id)
+                except ValueError:
```

Comment:
Why exactly is there a `ValueError` thrown here? What case does this `try`...`except` block here?